src/audio/src/src.c

```diff
@@ -903,15 +903,51 @@ static int src_copy(struct comp_dev *dev)
 	return 0;
 }
 
+static int src_check_buffer_sizes(struct comp_data *cd,
+				  struct audio_stream *source_stream,
+				  struct audio_stream *sink_stream)
+{
+	struct src_stage *s1 = cd->src.stage1;
+	struct src_stage *s2 = cd->src.stage2;
+	int stage1_times;
+	int stage2_times;
+	int blk_in;
+	int blk_out;
+	int n;
+
+	if (s2->filter_length > 1) {
+		/* Two polyphase filters case */
+		stage2_times = ceil_divide(cd->sink_frames, s2->blk_out);
+		stage1_times = ceil_divide(cd->source_frames, s1->blk_in);
+		blk_in = stage1_times * s1->blk_in;
+		blk_out = stage2_times * s2->blk_out;
+	} else {
+		/* Single polyphase filter case */
+		stage1_times = ceil_divide(cd->sink_frames, s1->blk_out);
+		n = ceil_divide(cd->source_frames, s1->blk_in);
+		stage1_times = MAX(stage1_times, n);
+		blk_in = stage1_times * s1->blk_in;
+		blk_out = stage1_times * s1->blk_out;
+	}
+
+	n = audio_stream_frame_bytes(source_stream) * (blk_in + cd->source_frames);
+	if (source_stream->size < n)
+		return -EIO;
+
+	n = audio_stream_frame_bytes(sink_stream) * (blk_out + cd->sink_frames);
+	if (source_stream->size < n)
+		return -EIO;
```

Comment:
wouldn't `-ENOBUFS` be a better error code?

Reply:
Done